src/fwupd_test.py

```diff
@@ -0,0 +1,345 @@
+        os.environ["DBUS_SYSTEM_BUS_ADDRESS"] = cls.test_bus.get_bus_address()
+
+        cls.dbus = Gio.bus_get_sync(Gio.BusType.SYSTEM, None)
+        cls.dbus_con = cls.get_dbus(True)
+
+        override_gi_search_path()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.test_bus.down()
+        dbusmock.DBusTestCase.tearDownClass()
+
+    def setUp(self):
+        self.testbed = UMockdev.Testbed.new()
+        self.polkitd, self.obj_polkit = self.spawn_server_template(
+            "polkitd", {}, stdout=subprocess.PIPE
+        )
+        self.obj_polkit.SetAllowed(
+            [
+                "org.freedesktop.fwupd.update-internal",
+                "org.freedesktop.fwupd.update-internal-trusted",
+                "org.freedesktop.fwupd.downgrade-internal-trusted",
+                "org.freedesktop.fwupd.downgrade-internal",
+                "org.freedesktop.fwupd.update-hotplug-trusted",
+                "org.freedesktop.fwupd.update-hotplug",
+                "org.freedesktop.fwupd.downgrade-hotplug-trusted",
+                "org.freedesktop.fwupd.downgrade-hotplug",
+                "org.freedesktop.fwupd.device-unlock",
+                "org.freedesktop.fwupd.modify-config",
+                "org.freedesktop.fwupd.device-activate",
+                "org.freedesktop.fwupd.verify-update",
+                "org.freedesktop.fwupd.modify-remote",
+                "org.freedesktop.fwupd.set-approved-firmware",
+                "org.freedesktop.fwupd.self-sign",
+                "org.freedesktop.fwupd.get-bios-settings",
+                "org.freedesktop.fwupd.set-bios-settings",
+                "org.freedesktop.fwupd.fix-host-security-attr",
+                "org.freedesktop.fwupd.undo-host-security-attr",
+            ]
+        )
+
+        self.proxy = None
+        self.props_proxy = None
+        self.daemon_log = None
+        self.daemon = None
+        self.changed_properties = {}
+
+    def run(self, result=None):
+        super().run(result)
+        if not result or not self.daemon_log:
+            return
+        if len(result.errors) + len(result.failures) or os.getenv("VERBOSE"):
+            with open(self.daemon_log.name, encoding="utf-8") as tmpf:
+                sys.stderr.write("\nDAEMON log:\n")
+                sys.stderr.write(tmpf.read())
+                sys.stderr.write("\n")
+
+    def tearDown(self):
+        del self.testbed
+        self.stop_daemon()
+
+        if self.polkitd:
+            self.polkitd.stdout.close()
+            try:
+                self.polkitd.kill()
+            except OSError:
+                pass
+            self.polkitd.wait()
+
+        self.obj_polkit = None
+
+    #
+    # Daemon control and D-BUS I/O
+    #
+
+    def start_daemon(self):
+        """Start daemon and create DBus proxy.
+
+        When done, this sets self.proxy as the Gio.DBusProxy for power-profiles-daemon.
+        """
+        env = os.environ.copy()
+        env["G_DEBUG"] = "fatal-criticals"
```

Comment:
Not your problem, but I was thinking of enabling `G_DEBUG=fatal-criticals` for non-`SUPPORTED_BUILD` -- or maybe even for `SUPPORTED_BUILD` too. Insane?

Reply:
Actually that's already done in a lot of places.  Most the end user facing apps have: 

```
#ifndef SUPPORTED_BUILD
	/* make critical warnings fatal */
	(void)g_setenv("G_DEBUG", "fatal-criticals", FALSE);
#endif
```

Reply:
Indeed, apologies for the noise!